src/modules/users/CMakeLists.txt

```diff
@@ -6,6 +6,16 @@
 find_package(${qtname} ${QT_VERSION} CONFIG REQUIRED Core DBus Network)
 find_package(Crypt REQUIRED)
 
+# Check for crypt_gensalt
+if(Crypt_FOUND)
+    list(APPEND CMAKE_REQUIRED_LIBRARIES crypt)
+    include(CheckSymbolExists)
+    check_symbol_exists(crypt_gensalt crypt.h HAS_CRYPT_GENSALT)
+    if(HAS_CRYPT_GENSALT)
+        add_definitions(-DHAVE_CRYPT_GENSALT)
+    endif()
```

Comment:
Do this later, when you have a target to add the definitions to. Also, you don't need `-D` when adding definitions. CMake is smart enough to know how to add a definition for the compiler being used. So just drop these lines

Reply:
Then down below, add lines
```
if(Crypt_FOUND AND HAS_CRYPT_GENSALT)
target_compile_definitions(users_internal PRIVATE HAVE_CRYPT_GENSALT)
endif()
```

(Note that the rest of the CMake code shows all of the bad habits that I'm telling you not to follow .. don't use it as an example, really)

Reply:
So to make sure I have this right, the whole thing should look like:

```cmake
# Check for crypt_gensalt
if(Crypt_FOUND)
    set(_old_CRL "${CMAKE_REQUIRED_LIBRARIES}")
    list(APPEND CMAKE_REQUIRED_LIBRARIES crypt)
    include(CheckSymbolExists)
    check_symbol_exists(crypt_gensalt crypt.h HAS_CRYPT_GENSALT)
    set(CMAKE_REQUIRED_LIBRARIES "${_old_CRL}")
endif()

if(Crypt_FOUND AND HAS_CRYPT_GENSALT)
    target_compile_definitions(users_internal PRIVATE HAVE_CRYPT_GENSALT)
endif()
```

Reply:
Just right! The second `if()` goes lower in the file, because it refers to the target *users_internal*, which must exist by then.

Reply:
I think I understand. Thanks!